cmd/snap/cmd_run.go

```diff
@@ -465,12 +482,27 @@ func straceCmd() ([]string, error) {
 	}, nil
 }
 
-func runCmdUnderStrace(origCmd, env []string) error {
+func (x *cmdRun) runCmdUnderGdb(origCmd, env []string) error {
+	env = append(env, "SNAP_CONFINE_RUN_UNDER_GDB=1")
+
+	cmd := []string{"sudo", "-E", "gdb", "-ex=r", "-ex=catch exec", "-ex=cont", "--args"}
```

Comment:
Hope it's not too late, but it would be great to include a comment why this sequence. As I understand gdb will stop right after raise(SIGTRAP) in the shim, then 'catch exec' will trap the exec call done in the shim, lastly continue enables the thing to run further until the point the exec() call is caught.